src/components/search.js

```diff
@@ -142,14 +136,12 @@ class Search extends Emitter {
             };
         };
 
-        this.needleCount = 0;
-
         /**
          * @private
          */
         this.triggerHistory = (message, cb) => {
 
-            if (this.needleCount < this.config.limit) {
+            if (this.needleCount < this.config.limit || this.messagesBetweenTimetokens) {
```

Comment:
So ```limit``` is ignored if ```config.end && config.start``` is provided? Important for docs.

Reply:
@ianjennings yes, we discussed this back and forth with you and @madisonsmith. Because user can't predict how many messages will be there between two dates, so we just give him way to pull them in paging way.

Reply:
Can you add a note about this condition to the docs here:
https://github.com/pubnub/chat-engine/blob/master/src/components/chat.js#L584

Reply:
@ianjennings changes has been pushed to this branch... waiting for CI to complete.